MH-12829 Switch and rename event details tabs

KatrinIhler opened the pull request:
This PR switches the metadata and general tabs in the event details modal so that the metadata tab is in front. This accomplishes two things: 1. It's consistent with the series details modal, and 2. it makes more sense in my opinion to view the metadata first.

Additionally the general tab is renamed to publications tab since it contains publications and nothing else. For consistency a number of files and identifiers are also renamed. Because of naming conflicts (see discussion on list here) I thought it best to rename the resources for the assets tab to contain 'Asset' in their name, thus avoiding conflicts concerning 'eventPublicationsResource'. (I hope I didn't miss anything during my renaming spree, if I did, just comment and I'll fix it.)

See the screenshot for more details: 

![screenshot from 2018-05-07 16-46-19](https://user-images.githubusercontent.com/11960278/39708078-7c687210-5216-11e8-96bc-a5d242ab8067.png)

_This work is sponsored by ETH._

Commit:
MH-12829 Switch metadata and general tab

Put metadata tab first in event details to be consistent with series
details and move the general tab after the assets tab.

Commit:
MH-12829 Rename general tab to publications tab

Rename the general tab in the event details modal to publications tab
since it's more appropriate. Rename all files, resources and so on to be
consistent.

staubesv commented on etc/security/mh_default_org.xml:
```diff
@@ -53,7 +53,7 @@
     <sec:intercept-url pattern="/admin-ng/event/*/comments" method="GET" access="ROLE_ADMIN, ROLE_UI_EVENTS_DETAILS_COMMENTS_VIEW"/>
     <sec:intercept-url pattern="/admin-ng/event/*/comment/*" method="GET" access="ROLE_ADMIN, ROLE_UI_EVENTS_DETAILS_COMMENTS_VIEW"/>
-    <sec:intercept-url pattern="/admin-ng/event/*/general.json" method="GET" access="ROLE_ADMIN, ROLE_UI_EVENTS_DETAILS_GENERAL_VIEW" />
+    <sec:intercept-url pattern="/admin-ng/event/*/publications.json" method="GET" access="ROLE_ADMIN, ROLE_UI_EVENTS_PUBLICATIONS_GENERAL_VIEW" />
```
Remove "GENERAL"

KatrinIhler replied:
Thanks, fixed!

staubesv reviewed (changes requested):
Hi @KatrinIhler 

Just found one little typo (wrong role name in mh_default_org.xml). Except for this, lgtm.

KatrinIhler commented:
@staubesv I had also missed to check in the changes I made to etc/listproviders/publication.channels.properties, fixed that, too.

staubesv commented:
Hi @KatrinIhler I'm currently testing this and found a first issue: If you click on the "Yes" in the Events table for a published event, the Event Detail modal opens the tab metadata. As the user clicks on a cell in the column "Published", he might expect the tab "Publications" to be opened.

Commit:
MH-12829 Fix opening event details when published

Open the publications tab instead of the first tab of the event details
modal when clicking on the 'Yes' in the 'Published' column for published
events.

KatrinIhler commented:
@staubesv This was broken already, but wasn't visible until now - fixed!
